core/src/machine/ppu.rs

```diff
@@ -0,0 +1,251 @@
+use crate::{
+    env::Display,
+    primitives::{Byte, Word, Memory},
+};
+
+
+/// Pixel processing unit.
+pub(crate) struct Ppu {
+    pub vram: Memory,
+    pub oam: Memory,
+
+    /// How many cycles did we already spent in this line?
+    cycle_in_line: u8,
```

Comment:
Perhaps it makes sense to extend the `CycleCounter` and use it here? Advantages could be to limit the range the value could be in.